### ISSUE TITLE
Does Bootstrap JS have a ready event (if not, would it be possible/useful)?

### ISSUE THREAD

**Issue by chrisblakley** [twbs/bootstrap#26885]
I'm introducing an option into my project that lets other developers choose whether to load the full Bootstrap resource (reboot, CSS, and bundled JS) or just the grid (reboot and grid CSS only).

It's working great, but I ran into a dependency issue on my end: my JavaScript file was set as a dependency of the Bootstrap JS so if my developers choose to only load the Bootstrap grid (without JS) my own JS file never loads.

I've taken some steps to decouple my JS, but what would make it so much better in this situation is if the Bootstrap JS file had an event trigger for when it was ready. I could then simply set up an event listener in my JS file for something like `bs.ready` and then run all of the dependent functions then.

Maybe there is another way of doing this that I am not thinking of, too. Looking through the Bootstrap JS file it appears that everything is running on DOM ready, so I could wait until window load to run my functions, but I'd like to avoid major front-end changes during the fragile UX time between DOM ready and window load when the user can partially interact with the site.

If this would be useful I'd be happy to make a pull request, but I'd want to make sure I'm not missing something. Seems like we could add a `$$$1(document).trigger('bs.ready');` at the end of the Bootstrap JS file, but again- maybe there is a better way to do it that I'm not considering.

**Comment by Johann-S** [comment]
We expose a global var in v4 it's `bootstrap` so if you check the presence of this var you'll be able to know if Bootstrap is present or not.

So a code like a that should do what you want: 
```js
$(function() {
  if (bootstrap) {
    // your code
  }
})

**Comment by chrisblakley** [comment]
Thanks- I appreciate the response. I did consider checking for `bootstrap`, but the only subtle difference is that I can't necessarily tell the difference between if Bootstrap is not ready yet vs. if it will not be loaded at all.

Like I said, I've decoupled my functions enough so that I don't *think* it'll be an issue, but I thought it'd be nice if the library could explicitly communicate when it was ready.
